lib/command/Command.js

```diff
@@ -156,13 +156,20 @@ class Command {
         return true;
     }
 
-    process(args, msg) {
+    process(args, msg, client) {
         if(!this.permissionCheck(msg)) {
             if(this.permissionMessage) {
                 msg.channel.createMessage(this.permissionMessage);
             }
             return;
         }
+        if (this.deleteCommand && msg.channel.guild && msg.channel.permissionsOf(client.user.id).has('manageMessages')) {
+            let originalExec = this.execute;
+            this.execute = (msg, args) => {
```

Comment:
If the main command has `deleteCommand: true` but the subcommand has `deleteCommand: false` the message will still be deleted even if the user calls the subcommand since the main command is processed before the subcommand.

I guess I could add the delete check before execute is called.